watchtower/wtdb/client_db.go

```diff
@@ -2231,23 +2237,40 @@ func (c *ClientDB) DeleteCommittedUpdate(id *SessionID, seqNum uint16) error {
 		}
 
 		// If the commits sub-bucket doesn't exist, there can't possibly
-		// be a corresponding update to remove.
+		// be corresponding updates to remove.
 		sessionCommits := sessionBkt.NestedReadWriteBucket(
 			cSessionCommits,
 		)
 		if sessionCommits == nil {
-			return ErrCommittedUpdateNotFound
+			return nil
 		}
 
-		var seqNumBuf [2]byte
-		byteOrder.PutUint16(seqNumBuf[:], seqNum)
+		// errFoundUpdates is an error we will use to exit early from
+		// the ForEach loop. The return of this error means that at
+		// least one committed update exists.
+		var errFoundUpdates = fmt.Errorf("found committed updates")
+		err := sessionCommits.ForEach(func(k, v []byte) error {
+			return errFoundUpdates
+		})
+		switch {
+		// If an expected error is returned, return that error.
+		case err != nil && !errors.Is(err, errFoundUpdates):
+			return err
 
-		if sessionCommits.Get(seqNumBuf[:]) == nil {
-			return ErrCommittedUpdateNotFound
+		// If no error is returned then the ForEach call back was never
+		// entered meaning that there are no un-acked committed updates.
+		// So we can exit now as there is nothing left to do.
+		case err == nil:
+			return nil
+
+		// If the errFoundUpdates signal error was returned then there
+		// are some updates that need to be deleted.
+		case errors.Is(err, errFoundUpdates):
```

Comment:
nit: move this to the beginning, then the `err != nil` case can just become the `default` case?